src/internal_events/aws_s3.rs

```diff
@@ -81,33 +84,93 @@ pub mod source {
     }
 
     #[derive(Debug)]
-    pub(crate) struct SqsMessageDeleteSucceeded<'a> {
-        pub message_id: &'a str,
+    pub(crate) struct SqsMessageDeleteSucceeded {
+        pub message_ids: Vec<DeleteMessageBatchResultEntry>,
     }
 
-    impl<'a> InternalEvent for SqsMessageDeleteSucceeded<'a> {
+    impl InternalEvent for SqsMessageDeleteSucceeded {
         fn emit_logs(&self) {
-            trace!(message = "Deleted SQS message.", %self.message_id);
+            let message_ids = self
+                .message_ids
+                .iter()
+                .map(|x| x.id.to_string())
+                .collect::<Vec<_>>()
+                .join(", ");
+            trace!(message = "Deleted SQS message(s).", %message_ids);
```

Comment:
Is there any way to only generate `message_ids` here if tracing is actually enabled? This seems like an expensive computation to throw away when not tracing.

Reply:
Addressed in ecb473acd.  Moving the expression to the `trace!` call itself avoids evaluating it unless the span would ultimately be accepted/dealt with/whatever-you-want-to-call it by the installed subscriber.